examples/SimonSays/src/SimonSays.c

```diff
@@ -0,0 +1,267 @@
+ ** @{ */
+/** \addtogroup Examples CIAA Firmware Examples
+ ** @{ */
+/** \addtogroup Games Simon example source file
+ ** @{ */
+
+/*
+ * Initials     Name
+ * ---------------------------
+ * AA         Alvaro Alonso
+ */
+
+ /*
+ * modification history (new versions first)
+ * -----------------------------------------------------------
+ * 20160607 v0.0.1   AA   first functional version
+ */
+
+/*==================[inclusions]=============================================*/
+#include "SimonSays.h"
+
+/*==================[macros and definitions]=================================*/
+
+/*==================[internal data declaration]==============================*/
+
+/*==================[internal functions declaration]=========================*/
+
+/*==================[internal data definition]===============================*/
+/** \brief File descriptor for digital output ports
+ *
+ * Device path /dev/dio/out/0
+ */
+static int32_t leds_fd;
+/** \brief File descriptor for digital input ports
+ *
+ * Device path /dev/dio/in/0
+ */
+static int32_t buttons_fd;
+/** \brief Current state of the States Machine
+ * Initial State or state after reset is IDLE
+ * IDLE: The machine is waiting for an input 
+ */
+static states state = IDLE;
+/** \brief Array of the sequence to be displayed by the leds **/
+static const uint8_t sequence[]={1, 2, 0, 1, 3, 3, 1, 2, 1, 0, 1, 1, 0, 1, 2, 2, 3, 3, 1, 1, 2, 1, 1, 3, 3, 3, 1, 0, 2, 1, 0, 0, 3, 0, 2, 1, 3, 2, 0, 3, 2, 0, 2, 2, 0, 2, 2, 1, 0, 0, 3, 2, 2, 2, 2, 1, 2, 2, 2, 0, 2, 2, 3, 2, 2, 1, 3, 2, 3, 2, 3, 3, 1, 1, 2, 0, 0, 0, 2, 1, 1, 2, 1, 3, 2, 0, 3, 3, 3, 2, 3, 3, 0, 0, 0, 0, 2, 3, 0, 1};
+static uint8_t level=0;
+
+/*==================[external data definition]===============================*/
+
+/*==================[internal functions definition]==========================*/
+
+/*==================[external functions definition]==========================*/
+/** \brief Main function
+ *
+ * This is the main entry point of the software.
+ *
+ * \returns 0
+ *
+ * \remarks This function never returns. Return value is only to avoid compiler
+ *          warnings or errors.
+ */
+int main(void)
+{
+   /* Starts the operating system in the Application Mode 1 */
+   /* This example has only one Application Mode */
+   StartOS(Normal);
+
+   /* StartOs shall never returns, but to avoid compiler warnings or errors
+    * 0 is returned */
+   return 0;
+}
+
+/** \brief Error Hook function
+ *
+ * This fucntion is called from the os if an os interface (API) returns an
+ * error. Is for debugging proposes. If called this function triggers a
+ * ShutdownOs which ends in a while(1).
+ *
+ * The values:
+ *    OSErrorGetServiceId
+ *    OSErrorGetParam1
+ *    OSErrorGetParam2
+ *    OSErrorGetParam3
+ *    OSErrorGetRet
+ *
+ * will provide you the interface, the input parameters and the returned value.
+ * For more details see the OSEK specification:
+ * http://portal.osek-vdx.org/files/pdf/specs/os223.pdf
```

Comment:
todo esto parece copy and paste de otro ejemplo, si es así deberian estar todos los copyright holders. :)